filebeat/harvester/forwarder.go

```diff
@@ -0,0 +1,81 @@
+package harvester
+
+import (
+	"errors"
+
+	"github.com/elastic/beats/filebeat/util"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/processors"
+)
+
+// Outlet interface is used for forwarding events
+type Outlet interface {
+	SetSignal(signal <-chan struct{})
+	OnEventSignal(data *util.Data) bool
+	OnEvent(data *util.Data) bool
+}
+
+// Forwarder contains shared options between all harvesters needed to forward events
+type Forwarder struct {
+	Config     ForwarderConfig
+	Outlet     Outlet
+	Processors *processors.Processors
+}
+
+// ForwarderConfig contains all config options shared by all harvesters
+type ForwarderConfig struct {
+	common.EventMetadata `config:",inline"`      // Fields and tags to add to events.
+	InputType            string                  `config:"input_type"`
+	Pipeline             string                  `config:"pipeline"`
+	Module               string                  `config:"_module_name"`  // hidden option to set the module name
+	Fileset              string                  `config:"_fileset_name"` // hidden option to set the fileset name
+	Processors           processors.PluginConfig `config:"processors"`
+}
+
+func NewForwarder(cfg *common.Config, outlet Outlet) (*Forwarder, error) {
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported function NewForwarder should have comment or be unexported